src/test/java/org/folio/inventory/validation/experimental/TargetItemStatusValidatorTest.java

```diff
@@ -0,0 +1,81 @@
+package org.folio.inventory.validation.experimental;
+
+import org.assertj.core.util.Arrays;
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemStatusName;
+
+import org.folio.inventory.domain.items.Status;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+public class TargetItemStatusValidatorTest {
+  private static TargetItemStatusValidator validator;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    validator = new TargetItemStatusValidator();
+  }
+
+  @Test
+  public void inProcesAllowMarkAs() {
+    ItemStatusName targetItemStatusName = ItemStatusName.IN_PROCESS;
+    final AbstractTargetValidator statusValidator = validator.getValidator(targetItemStatusName);
+
+    Set<ItemStatusName> allowedStatuses = statusValidator.getAllStatusesAllowedToMark();
+    System.out.println("Allowed statuses for:"+statusValidator.getItemStatusName());
+    allowedStatuses.stream().forEach(x -> {
+      System.out.println("\t"+x);
+      Item item = new Item(null, null, new Status(x), null, null, null);
+      assertTrue(statusValidator.isItemAllowedToMark(item));
+    });
```

Comment:
This code seems to check that the statuses that production code thinks should be allowed are allowed by the validator.

Have I understood this correctly?

Reply:
Sure

Reply:
Apologies for the delayed response, I didn't realise you had replied to this comment.

How does it test the workflow?

From my perspective, this test only checks that the production code is consistent with itself. 

I think tests should demonstrate the intended behaviour rather than what the code says it does. As an example, if a status was incorrectly removed from the production code, this test would still pass. It does not raise awareness of the impact of a change or that something might have been incorrectly changed. 

Does that make sense?

I also think that tests should act as documentation for the behaviour of the system. When I read this test, it tells me that the validator will allow a status change when the source status is in the set defined by the system. I would find it more useful for understanding the intended behaviour if it described which of those statuses were allowed.

What do you think?

Below is an example of a test that I think better describes the intended behaviour of the system:
```
  @SneakyThrows
  @Parameters({
    "Available",
    "In transit",
    "Awaiting pickup",
    "Missing",
    "Withdrawn",
    "Lost and paid",
    "Paged",
    "Awaiting delivery",
    "Order closed"
  })
  @Test
  public void itemCanBeMarkedAsInProcessWhenInAcceptableSourceStatus(String sourceStatus) {
    final var validator = getValidator(IN_PROCESS);

    final var item = itemInStatus(forName(sourceStatus));

    final var validationFuture = validator.itemHasAllowedStatusToMark(item);

    validationFuture.get(1, TimeUnit.SECONDS);

    // Validator responds with a successful future when valid
    assertThat(validationFuture.isDone(), is(true));
  }
```

Reply:
Marc, you are right. I thoght more about main code structure than about tests. I think it would be great way if we could to take Item statuses and Transitions some where else where current approved reqiremnts would stored. But right now Statuses and allowed Transitions statically saved in test is a single way despite it time consuming and weak informativity (i mean you should review all tests and validators files to have a full picture of workflow). Also. I think,  I can mock MoveItemToStatusService class and make tests on it. Does it make sence?

Reply:
>  I think it would be great way if we could to take Item statuses and Transitions some where else where current approved requirements would stored.

I'm not sure I understand what you mean. Please could you explain this more.

> right now Statuses and allowed Transitions statically saved in test is a single way despite it time consuming and weak informativity

What part of this do you consider time consuming?

What do you mean by `weak informativity`?

> I think, I can mock MoveItemToStatusService class and make tests on it. Does it make sense?

I'm ok with that if you want to. If there is behaviour in that class that the API tests do not cover then please do.